graphx/src/main/scala/org/apache/spark/graphx/lib/PageRank.scala

```diff
@@ -196,11 +196,45 @@ object PageRank extends Logging {
     require(sources.nonEmpty, s"The list of sources must be non-empty," +
```

Comment:
You can remove these checks if they are now checked in the other method.

Reply:
Done